python/cudf/cudf/core/reshape.py

```diff
@@ -642,67 +658,79 @@ def get_dummies(
     if drop_first:
         raise NotImplementedError("drop_first is not supported yet")
 
-    encode_fallback_dtypes = ["object", "category"]
+    if isinstance(df, cudf.DataFrame):
+        encode_fallback_dtypes = ["object", "category"]
 
-    if columns is None or len(columns) == 0:
-        columns = df.select_dtypes(include=encode_fallback_dtypes).columns
+        if columns is None or len(columns) == 0:
+            columns = df.select_dtypes(include=encode_fallback_dtypes).columns
 
-    def length_check(obj, name):
-        if cudf.utils.dtypes.is_list_like(obj):
-            if len(obj) != len(columns):
-                raise ValueError(
-                    f"Length of '{name}' ({len(obj)}) did not match the "
-                    f"length of the columns being encoded ({len(columns)})."
-                )
+        def length_check(obj, name):
+            if cudf.utils.dtypes.is_list_like(obj):
+                if len(obj) != len(columns):
+                    raise ValueError(
+                        f"Length of '{name}' ({len(obj)}) did not match the "
+                        f"length of the columns being "
+                        f"encoded ({len(columns)})."
+                    )
 
-    length_check(prefix, "prefix")
-    length_check(prefix_sep, "prefix_sep")
+        length_check(prefix, "prefix")
+        length_check(prefix_sep, "prefix_sep")
 
-    if prefix is None:
-        prefix = columns
+        if prefix is None:
+            prefix = columns
 
-    if isinstance(prefix, str):
-        prefix_map = {}
-    elif isinstance(prefix, dict):
-        prefix_map = prefix
-    else:
-        prefix_map = dict(zip(columns, prefix))
+        if isinstance(prefix, str):
+            prefix_map = {}
+        elif isinstance(prefix, dict):
+            prefix_map = prefix
+        else:
+            prefix_map = dict(zip(columns, prefix))
 
-    if isinstance(prefix_sep, str):
-        prefix_sep_map = {}
-    elif isinstance(prefix_sep, dict):
-        prefix_sep_map = prefix_sep
-    else:
-        prefix_sep_map = dict(zip(columns, prefix_sep))
+        if isinstance(prefix_sep, str):
+            prefix_sep_map = {}
+        elif isinstance(prefix_sep, dict):
+            prefix_sep_map = prefix_sep
+        else:
+            prefix_sep_map = dict(zip(columns, prefix_sep))
+
+        # If we have no columns to encode, we need to drop
+        # fallback columns(if any)
+        if len(columns) == 0:
+            return df.select_dtypes(exclude=encode_fallback_dtypes)
+        else:
+            result_df = df.drop(columns=columns)
```

Comment:
I believe `.drop` returns via deep copy. Could we use a shallow copy and inplace drop here instead?

Reply:
Yep, made this change 👍